# 109B_Project

Data: We plan to use the annotation files from GENCODE, which can be accessed at
https://www.gencodegenes.org/human/. This database contains the gene sequences and their
functional annotations (exons, introns, etc.).

Background and Motivation: Innovations in deep learning opens the door to tremendous
opportunities in demystifying DNA sequences, the “blueprint” of life. However, the vast majority
of these sequences do not code for proteins and will be spliced out as “introns”, while the
remaining “exons” play a critical role in gene expression. Unraveling the functionality and
regulatory mechanisms of DNA sequences is pivotal for advances in genetic research,
medicine, and biotechnology.

Problem: Functional annotation of DNA sequences plays a major role in the field of
computational genomics. Our goal is to use convolutional and recurrent neural networks (CNN
& RNN) to classify DNA sequences into their corresponding functional categories, including
exon, intron, untranslated regions, and so on.

Scope and Methods: Our project aims to develop CNN and RNN-based deep learning models
to explore the functions of DNA sequences. We plan to explore different existing DNA modeling
architectures and techniques to decode the dictionary of life. Through this project, our team will
practice our knowledge of deep learning models and gain valuable hands-on experience.

Concerns & Limitations:
○ Ethical: The DNA sequence data in GENCODE is publicly accessible and widely
used by research initiatives.
○ Data Quality: A preliminary look at the GENCODE database indicates that
high-quality and standardized genomic annotations are provided, but the
sequences will need to be extracted from the annotation file.
○ Computational: Depending on the number of sequences used and the complexity
of the model architectures, the computational resources might be a challenge.
